langchain4j-core/src/main/java/dev/langchain4j/rag/DefaultRetrievalAugmentor.java

```diff
@@ -227,62 +225,69 @@ private static List<Content> retrieve(ContentRetriever retriever, Query query) {
         if (queries.size() == 1) {
             Query transformedQuery = queries.iterator().next();
             if (!transformedQuery.equals(originalQuery)) {
                 log.debug("Transformed original query '{}' into '{}'",
-                        originalQuery.text(), transformedQuery.text());
+                    originalQuery.text(), transformedQuery.text());
             }
-        } else {
+        } else if (log.isDebugEnabled()){
             log.debug("Transformed original query '{}' into the following queries:\n{}",
-                    originalQuery.text(), queries.stream()
-                            .map(Query::text)
-                            .map(query -> "- '" + query + "'")
-                            .collect(joining("\n")));
+                originalQuery.text(), queries.stream()
+                    .map(Query::text)
+                    .map(query -> "- '" + query + "'")
+                    .collect(joining("\n")));
         }
     }
 
     private static void log(Query query, Collection<ContentRetriever> retrievers) {
         // TODO use retriever id
         if (retrievers.size() == 1) {
             log.debug("Routing query '{}' to the following retriever: {}",
-                    query.text(), retrievers.iterator().next());
-        } else {
+                query.text(), retrievers.iterator().next());
+        } else if (log.isDebugEnabled()) {
             log.debug("Routing query '{}' to the following retrievers:\n{}",
-                    query.text(), retrievers.stream()
-                            .map(retriever -> "- " + retriever.toString())
-                            .collect(joining("\n")));
+                query.text(), retrievers.stream()
+                    .map(retriever -> "- " + retriever.toString())
+                    .collect(joining("\n")));
         }
     }
 
     private static void log(Query query, ContentRetriever retriever, List<Content> contents) {
         // TODO use retriever id
         log.debug("Retrieved {} contents using query '{}' and retriever '{}'",
-                contents.size(), query.text(), retriever);
+            contents.size(), query.text(), retriever);
 
-        if (!contents.isEmpty()) {
-            final var contentsSting = contents.stream()
+        if (log.isTraceEnabled()) {
+            if (!contents.isEmpty()) {
+                final var contentsSting = contents.stream()
                     .map(Content::textSegment)
                     .map(segment -> "- " + escapeNewlines(segment.text()))
                     .collect(joining("\n"));
-            log.trace("Retrieved {} contents using query '{}' and retriever '{}':\n{}",
+                log.trace("Retrieved {} contents using query '{}' and retriever '{}':\n{}",
                     contents.size(),
                     query.text(),
                     retriever.getClass().getName(),
                     contentsSting);
+            }
+        } else {
```

Comment:
looks like else belongs to wrong if branch here. Should it be for inner if?

Reply:
[Fixed](https://github.com/langchain4j/langchain4j/pull/1987/files#diff-9a5519a26a4b6d2fd412c877de4459c2a119e793fece9385f558a93a7e0aee5aR273-R275)